src/Container.tsx

```diff
@@ -1,87 +1,91 @@
+import DialogButton, { DialogButtonProps } from "./Button";
+
+type TitleElement = ReactElement<DialogTitleProps, typeof DialogTitle>;
+type DescriptionElement = ReactElement<
+  DialogDescriptionProps,
+  typeof DialogDescription
+>;
+type ButtonElement = ReactElement<DialogButtonProps, typeof DialogButton>;
 
 const iOS = Platform.OS === "ios";
 
-export interface DialogContainerProps {
+export type DialogContainerProps = PropsWithChildren<{
   blurComponentIOS?: ReactNode;
-  buttonSeparatorStyle?: ViewStyle;
-  contentStyle?: ViewStyle;
-  footerStyle?: ViewStyle;
-  headerStyle?: ViewStyle;
-  blurStyle?: ViewStyle;
+  buttonSeparatorStyle?: StyleProp<ViewStyle>;
+  contentStyle?: StyleProp<ViewStyle>;
+  footerStyle?: StyleProp<ViewStyle>;
+  headerStyle?: StyleProp<ViewStyle>;
+  blurStyle?: StyleProp<ViewStyle>;
   visible?: boolean;
   verticalButtons?: boolean;
   onBackdropPress?: () => void;
   onRequestClose?: () => void;
   keyboardVerticalOffset?: number;
-  children: ReactElement<any, NamedExoticComponent>[];
-}
+}>;
 
 const DialogContainer: React.FC<DialogContainerProps> = (props) => {
   const {
     blurComponentIOS,
-    buttonSeparatorStyle = {},
+    buttonSeparatorStyle,
     children,
-    contentStyle = {},
-    footerStyle = {},
-    headerStyle = {},
-    blurStyle = {},
+    contentStyle,
+    footerStyle,
+    headerStyle,
+    blurStyle,
     visible = false,
     verticalButtons = false,
     keyboardVerticalOffset = 40,
     ...nodeProps
   } = props;
-  const titleChildrens: ReactElement<any, NamedExoticComponent>[] = [];
-  const descriptionChildrens: ReactElement<any, NamedExoticComponent>[] = [];
-  const buttonChildrens: ReactElement<any, NamedExoticComponent>[] = [];
-  const otherChildrens: ReactElement<any, NamedExoticComponent>[] = [];
+  const titleChildrens: TitleElement[] = [];
+  const descriptionChildrens: DescriptionElement[] = [];
+  const buttonChildrens: (ButtonElement | JSX.Element)[] = [];
+  const otherChildrens: ReactNode[] = [];
   const { styles } = useTheme(buildStyles);
   React.Children.forEach(children, (child) => {
-    if (!child) {
-      return;
-    }
-    if (
-      child.type.name === "DialogTitle" ||
-      child.type.displayName === "DialogTitle"
-    ) {
-      titleChildrens.push(child);
-    } else if (
-      child.type.name === "DialogDescription" ||
-      child.type.displayName === "DialogDescription"
-    ) {
-      descriptionChildrens.push(child);
-    } else if (
-      child.type.name === "DialogButton" ||
-      child.type.displayName === "DialogButton"
-    ) {
-      if (Platform.OS === "ios" && buttonChildrens.length > 0) {
-        buttonChildrens.push(
-          <View
-            style={[
-              verticalButtons
-                ? styles.buttonSeparatorVertical
-                : styles.buttonSeparatorHorizontal,
-              buttonSeparatorStyle,
-            ]}
-          />
-        );
```

Comment:
Wow, I think I haven't [read this code in ages](https://github.com/mmazzarolo/react-native-dialog/blob/8cc69b78cf391a39994abc6774b439ac29622a9f/src/Container.js), it's 4 years old — no idea why the strings were hardcoded in the first place, haha. 
Thank you so much for cleaning this up!